dist/index.js

```diff
@@ -0,0 +1,183 @@
+    token = null;
+  }
+
+  if (token) {
+    debug('Token defined: %s…', token.substring(0, 6));
+    this.token = token;
+  }
+
+  // Set default request handler
+  if (!reqHandler) {
+    debug('No request handler. Adding default XHR request handler');
+
+    this.request = function (params, fn) {
+      params = params || {};
+
+      // token is optional
+      if (token) {
+        params.authToken = token;
+      }
+
+      return request_handler(params, fn);
+    };
+  } else {
+    this.request = reqHandler;
+  }
+
+  // Add Req instance
+  this.req = new Req(this);
+
+  // Default api version;
+  this.apiVersion = '1.1';
+}
+
+/**
+ * Get `Me` object instance
+ *
+ * @api public
+ */
+
+WPCOM.prototype.me = function () {
+  return new Me(this);
+};
+
+/**
+ * Get `Site` object instance
+ *
+ * @param {String} id
+ * @api public
+ */
+
+WPCOM.prototype.site = function (id) {
+  return new Site(id, this);
+};
+
+/**
+ * Get `Users` object instance
+ *
+ * @api public
+ */
+
+WPCOM.prototype.users = function () {
+  return new Users(this);
+};
+
+WPCOM.prototype.batch = function () {
+  return new Batch(this);
+};
+
+/**
+ * List Freshly Pressed Posts
+ *
+ * @param {Object} [query]
+ * @param {Function} fn callback function
+ * @api public
+ */
+
+WPCOM.prototype.freshlyPressed = function (query, fn) {
+  return this.req.get('/freshly-pressed', query, fn);
+};
+
+/**
+ * Expose send-request
+ * @TODO: use `this.req` instead of this method
+ */
+
+WPCOM.prototype.sendRequest = function (params, query, body, fn) {
+  var msg = 'WARN! Don use `sendRequest() anymore. Use `this.req` method.';
+  if (console && console.warn) {
+    //eslint-disable-line no-console
+    console.warn(msg); //eslint-disable-line no-console
+  } else {
+      console.log(msg); //eslint-disable-line no-console
+    }
+
+  return sendRequest.call(this, params, query, body, fn);
+};
+
+if (!Promise.prototype.timeout) {
+  /**
+      * Returns a new promise with a deadline
+      *
+      * After the timeout interval, the promise will
+      * reject. If the actual promise settles before
+      * the deadline, the timer is cancelled.
+      *
+      * @param {number} delay how many ms to wait
+      * @returns {Promise}
+      */
+  Promise.prototype.timeout = function () {
+    var _this = this;
+
+    var delay = arguments.length <= 0 || arguments[0] === undefined ? DEFAULT_ASYNC_TIMEOUT : arguments[0];
+
+    var cancelTimeout = undefined,
+        timer = undefined,
+        timeout = undefined;
+
+    timeout = new Promise(function (resolve, reject) {
+      timer = setTimeout(function () {
+        reject(new Error('Action timed out while waiting for response.'));
+      }, delay);
+    });
+
+    cancelTimeout = function () {
+      clearTimeout(timer);
+      return _this;
+    };
+
+    return Promise.race([this.then(cancelTimeout)['catch'](cancelTimeout), timeout]);
+  };
+}
+
+/**
+ * Expose `WPCOM` module
+ */
+
+module.exports = WPCOM;
```

Comment:
Taken care of in https://github.com/Automattic/wpcom.js/commit/4a1cb85c54a3f55e961068d4479ebfba0e589314